### PR TITLE
Feat(BB-86): Prepopulate entity editor with POST /create

### PR DESCRIPTION
**Pull request by tr1ten**
### Problem

BB-86: prepopulate creation forms via POST route

### Solution

Add /create post route for each entity to prefill their entity editor form with post data.

### Routes 
- [x] Edition
- [x] Publisher
- [x] Edition-Group
- [x] Series
- [x] Author
- [x] Work  
### Areas of Impact

entity routes and common utils

### CODE REVIEW

**Review by MonkeyDo — COMMENTED** [review]
This is really exciting, nice work so far !
I've only doing a quick review, everything seems to be working pretty well altogether.
Adding some basic tests would be ideal, and would allow us to throw curveballs at the routes and ensure nothing breaks.

One thing that is currently missing: the duplicate detection mechanism (detects same name, triggered on input change)  isn't run when we load the page at first, which is absolutely required to prevent people adding lots of duplicates.
I haven't dug into how to trigger it on page load with the pre-filled data, let me know if you need help with that.


And here's an odd one: I've run into this error in some cases:
```
TypeError: (0 , I.entityToOption) is not a function
    at entity-editor.js:formatted:1576:42
    at Array.map (<anonymous>)
    at dn (entity-editor.js:formatted:1571:22)
```
    
It comes from this line: https://github.com/metabrainz/bookbrainz-site/blob/master/src/client/entity-editor/edition-section/edition-section.tsx#L293
I'm a bit baffled; why would entityToOption not be a function !?

To reproduce: Try to import a book which is already in the DB (or import it twice I guess)
Now ad a space or some small change to the name, and remove it.
This triggers the mechanism that detects possible duplicates.
Page goes blank with the above error in the console.

File: src/server/routes/entity/edition.js [hunk 1]
```diff
@@ -212,6 +212,56 @@ router.get(
...
+	'/create', entityRoutes.displayPreview, auth.isAuthenticatedForHandler, middleware.loadIdentifierTypes,
+	middleware.loadEditionStatuses, middleware.loadEditionFormats,
+	middleware.loadLanguages, middleware.loadRelationshipTypes,
+	async (req, res, next) => {
+		// parsing submitted data to correct format
+		const entity = await utils.parseInitialState(req);
+		if (entity.editionSection) {
+			const {orm} = req.app.locals;
+			const {EditionFormat} = orm;
+			entity.editionSection = await utils.parseLanguages(entity.editionSection, orm);
```
[the remark this answers is not in the document]
Comment (tr1ten) [reply src/server/routes/entity/edition.js:225]: Edition section languages working fine for me, please make sure given format is correct (like editionSection.languages1: English).

**Comment by MonkeyDo** [comment]
Well, I'm just a big ol' dummy, I'm the one who broke that import of the entityToOption utility: https://github.com/metabrainz/bookbrainz-site/commit/365faec58d41422e21cc62894e365e1f2eed5e4b

Sorry about that ! I'll make a correction and update this PR once it's in the master branch

**Comment by MonkeyDo** [comment]
I've updated test.bookbrainz.org after merging the fix.

**Review by MonkeyDo — COMMENTED** [review]
All righty, final stretch !

I'll deploy the PR again on test.BB

**Review by MonkeyDo — APPROVED** [review]
All right ! Thanks for all the hard work !

As a logical continuation, I'm looking forward to shifting our focus to the userscripts

**Pull request merged**